docs/v1/p4runtime-id-notes.md

```diff
@@ -0,0 +1,487 @@
++ `GetForwardingPipelineConfigRequest`
++ `WriteRequest`
++ `ReadRequest`
+
+Here are two examples of how a P4Runtime server implementation might
+select `device_id` values:
+
++ The values could be compiled into the P4Runtime server's code at
+  compile time.
++ The P4Runtime server could do some kind of device discovery on a
+  hardware board to determine how many devices there are, and assign
+  then numbers in some predictable way, e.g. from 1 up to the number
+  of devices.
+
+Other ways are possible.  These examples are only mentioned to give
+some idea of how they might be allocated.
+
+
+### Role ids
+
+Scope: TBD
+
+`role_id` field values appear in the message types listed below.
+Their values are selected by clients.  The P4Runtime API specification
+does not define the meaning of any numeric values of role id, with the
+exception of 0.  A role id of 0 is always a role that can modify
+anything that is modifiable via the P4Runtime API.
+
++ `Role`, where the field is called `id` instead of `role_id`
+  + `MasterArbitrationUpdate` messages contain `Role` sub-messages
++ `SetForwardingPipelineConfigRequest`
++ `WriteRequest`
+
+There is no known use case for a P4 data plane to know role id values.
+
+
+### Election ids
+
+Scope: A single P4Runtime device.
+
+`election_id` field values appear in the message types listed below.
+Their values are selected by P4Runtime clients.
+
++ `MasterArbitrationUpdate`
++ `SetForwardingPipelineConfigRequest`
++ `WriteRequest`
+
+There is no known use case for a P4 data plane to know election id
+values.
+
+
+### Multicast group ids
+
+Scope: A single P4Runtime device.
+
+`multicast_group_id` field values appear in the message types listed
+below.  The allowed set of values is determined by the P4Runtime
+server, typically in a contiguous range `[1, M]` for some maximum
+number of multicast groups `M` supported by the device, which can
+differ from one device to another.
+
+A particular multicast group id value X (e.g. X=7) could have a
+different replication list for every device.
+
++ `MulticastGroupEntry`
+
+Multicast group id values must be used by the P4 data plane to direct
+packets to be replicated using the replication list configured for
+that multicast group id.  Thus it is likely that a P4Runtime client
+would first configure a multicast group id, then write that id into a
+place readable by the P4 program, e.g. as a parameter of some action
+of a table.
+
+(It is also possible that a P4 program might have some multicast group
+id values compiled into it for use for particular purposes, in which
+case it is up to some agreement between the developer of the P4
+program and the controller software.)
+
+
+### Clone session ids
+
+Scope: A single P4Runtime device.
+
+`session_id` field values appear in the message types listed below.
+The allowed set of values is determined by the P4Runtime server,
+typically in a contiguous range `[0, C-1]` for some maximum number of
```

Comment:
0 is no longer valid since P4Runtime v1.1.0

Reply:
Thanks for catching that.  Changed to `[1, C]`